src/main/java/com/hierynomus/smbj/share/PipeShare.java

```diff
@@ -0,0 +1,107 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hierynomus.smbj.share;
+
+import com.hierynomus.msdtyp.AccessMask;
+import com.hierynomus.mserref.NtStatus;
+import com.hierynomus.msfscc.FileAttributes;
+import com.hierynomus.msfscc.fsctl.FsCtlPipeWaitRequest;
+import com.hierynomus.mssmb2.SMB2CreateDisposition;
+import com.hierynomus.mssmb2.SMB2CreateOptions;
+import com.hierynomus.mssmb2.SMB2FileId;
+import com.hierynomus.mssmb2.SMB2ShareAccess;
+import com.hierynomus.mssmb2.messages.SMB2CreateResponse;
+import com.hierynomus.mssmb2.messages.SMB2IoctlResponse;
+import com.hierynomus.smbj.common.SMBApiException;
+import com.hierynomus.smbj.common.SMBBuffer;
+import com.hierynomus.smbj.common.SmbPath;
+import com.hierynomus.smbj.io.ArrayByteChunkProvider;
+
+import java.util.Set;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+public class PipeShare extends Share {
+    private static final int FSCTL_PIPE_WAIT                         = 0x00110018;
+
+    public PipeShare(SmbPath smbPath, TreeConnect treeConnect) {
+        super(smbPath, treeConnect);
+    }
+
+    /**
+     * Requests that the server wait until an instance of the specified named pipe is available for connection.
+     * <p>
+     * Name must not include the "\pipe\", so if the operation was on \\server\pipe\pipename, the name would be "pipename".
+     * <p>
+     * This method requests that the server wait indefinitely. To specify a maximum wait time use {@link #waitForPipe(String, long, TimeUnit)}.
+     *
+     * @param name the name of the named pipe.
+     * @return true if an instance of the pipe is available; false if a timeout occurred
+     * @throws SMBApiException if an error occurs while waiting for an instance of the pipe to become available
+     */
+    public boolean waitForPipe(String name) {
+        return waitForPipe(name, 0, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Requests that the server wait until an instance of the specified named pipe is available for connection.
+     * <p>
+     * Name must not include the "\pipe\", so if the operation was on \\server\pipe\pipename, the name would be "pipename".
+     *
+     * @param name        the name of the named pipe.
+     * @param timeout     the amount of time to wait until an instance is available
+     * @param timeoutUnit the unit in which timeout is specified
+     * @return true if an instance of the pipe is available; false if a timeout occurred
+     * @throws SMBApiException if an error occurs while waiting for an instance of the pipe to become available
+     */
+    public boolean waitForPipe(String name, long timeout, TimeUnit timeoutUnit) {
+        long timeoutMs = timeoutUnit.toMillis(timeout);
+
+        SMBBuffer buffer = new SMBBuffer();
+        new FsCtlPipeWaitRequest(name, timeoutMs / 100, timeoutMs > 0).write(buffer);
```

Comment:
I'm not the one that makes this stuff up 😄 See https://msdn.microsoft.com/en-us/library/cc232126.aspx.

> Timeout (8 bytes): A 64-bit signed integer that specifies the maximum amount of time, in units of 100 milliseconds, that the function can wait for an instance of the named pipe to be available.